theano/sandbox/cuda/basic_ops.py

```diff
@@ -4288,3 +4289,100 @@ def c_code(self, node, name, inp, out, sub):
     def c_code_cache_version(self):
         return (3,)
 gpu_eye = GpuEye(dtype='float32')
+
+
+class GpuDiagonal(GpuOp):
+    __props__ = ("offset", "axis1", "axis2")
```

Comment:
Yes it is. When we compare two ```GpuDiagonal```s, if one has ```view=False``` and the other has ```view=True```, then they have different behavior and should be considered different by ```__eq__()```